### PR TITLE
Fix `__eq__` method of S.Reals

### PR DESCRIPTION
**Pull request by aktech**
Currently `S.Reals == Interval(0, oo)` returns `None`.

### CODE REVIEW

**Comment by aktech** [comment]
@hargup

**Comment by hargup** [comment]
+1

File: sympy/sets/fancysets.py [hunk 1]
```diff
@@ -179,8 +179,7 @@ def __new__(cls):
         return Interval.__new__(cls, -S.Infinity, S.Infinity)
 
     def __eq__(self, other):
-        if other == Interval(-S.Infinity, S.Infinity):
-            return True
+        return other == Interval(-S.Infinity, S.Infinity)
```
Comment (asmeurer) [inline sympy/sets/fancysets.py]: What about `S.Reals == S.Reals`?

Comment (aktech) [inline sympy/sets/fancysets.py]: That would eventually boil down to `S.Reals == Interval(-oo, oo)`.

**Comment by aktech** [comment]
Thanks!

**Pull request merged**